pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -225,6 +240,48 @@ func (mutator *VMsMutator) setDefaultPreferenceKind(vm *v1.VirtualMachine) {
 	}
 }
 
+func validateInstancetypeMatcherUpdate(oldInstancetypeMatcher *v1.InstancetypeMatcher, newInstancetypeMatcher *v1.InstancetypeMatcher) []metav1.StatusCause {
+	if oldInstancetypeMatcher == nil && newInstancetypeMatcher == nil {
+		return nil
+	}
+	if err := validateMatcherUpdate(*oldInstancetypeMatcher, *newInstancetypeMatcher); err != nil {
+		return []metav1.StatusCause{{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: err.Error(),
+			Field:   k8sfield.NewPath("spec", "instancetype", "revisionName").String(),
+		}}
+	}
+	return nil
+}
+
+func validatePreferenceMatcherUpdate(oldPreferenceMatcher *v1.PreferenceMatcher, newPreferenceMatcher *v1.PreferenceMatcher) []metav1.StatusCause {
+	if oldPreferenceMatcher == nil && newPreferenceMatcher == nil {
+		return nil
+	}
+	if err := validateMatcherUpdate(*oldPreferenceMatcher, *newPreferenceMatcher); err != nil {
+		return []metav1.StatusCause{{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: err.Error(),
+			Field:   k8sfield.NewPath("spec", "preference", "revisionName").String(),
+		}}
+	}
+	return nil
+}
+
+func validateMatcherUpdate[M v1.Matcher](oldMatcher M, newMatcher M) error {
+	// Do not check anything when the original matcher didn't have a revisionName as this is likely the VM Controller updating the matcher
+	if oldMatcher.GetRevisionName() == "" {
+		return nil
+	}
+	// If the matchers have changed ensure that the RevisionName is cleared when updating the Name
+	if !equality.Semantic.DeepEqual(newMatcher, oldMatcher) {
+		if oldMatcher.GetName() != newMatcher.GetName() && newMatcher.GetRevisionName() != "" {
+			return fmt.Errorf("the Matcher Name has been updated without clearing the RevisionName")
+		}
```

Comment:
Change this to

```
		if oldMatcher.GetName() != newMatcher.GetName() && oldMatcher.GetRevisionName() == newMatcher.GetRevisionName() {
			return fmt.Errorf("the Matcher Name has been updated without updating the RevisionName")
		}
```

to allow changing to a different matcher with a specific revision in one go?